### PR TITLE
caom2pipe refactoring

### PR DESCRIPTION
**Pull request by SharonGoliath**
Extract the transfer_composable.py functionality from the execute_composable.py implementation, to reduce the number of variations on CaomExecute.

Move functions from execute_composable.py to classes in run_composable.py.

### CODE REVIEW

File: caom2pipe/execute_composable.py [hunk 1]
```diff
@@ -552,66 +420,21 @@ class MetaUpdateClient(CaomExecute):
...
-        self.observation = observation
-
-    def execute(self, context):
-        self.logger.debug(f'Begin execute for {self.__class__.__name__}')
+        self.logger.debug('Begin execute')
         self.logger.debug('the steps:')
 
-        self.logger.debug('Find the file name as stored.')
-        self._cadc_data_info_file_name_client()
+        # TODO - what breaks when this isn't here? But it can't be here,
```
Comment (andamian) [inline caom2pipe/execute_composable.py:433]: Is this TODO something that needs attention? I've noticed it in several places.

File: caom2pipe/execute_composable.py [hunk 2]
```diff
@@ -1729,184 +1142,7 @@ def choose(self, storage_name, command_name, meta_visitors, data_visitors):
...
-                    pass
-                else:
-                    raise mc.CadcException(
-                        f'Do not understand task type {task_type}')
-        else:
-            logging.error(f'{storage_name.obs_id} failed naming validation '
-                          f'check.')
-            self.capture_failure(storage_name, 'Invalid observation ID')
-        return executors
+        pass
```
Comment (andamian) [inline caom2pipe/execute_composable.py:1145]: Method needed or not?

File: caom2pipe/manage_composable.py [hunk 3]
```diff
@@ -469,10 +434,10 @@ def capture(self):
             create_dir(self.observable_dir)
             now = datetime.utcnow().timestamp()
             for service in self.history.keys():
-                fqn = '{}/{}.{}.yml'.format(self.observable_dir, now, service)
+                fqn = f'{self.observable_dir}/{now}.{service}.yml'
```
Comment (andamian) [inline caom2pipe/manage_composable.py:437]: I tend to use `os.path.join(dir, filename)` just to make the code portable. I know it's not the case here but it's a good practice.

File: caom2pipe/transfer_composable.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    def observable(self, value):
+        self._observable = value
+
+    def get(self, source, dest_fqn):
+        """
+        This no-op implementation is used when use_local_files = True.
+        :param source:
+        :param dest_fqn: str - file-system based fully-qualified name
+        """
+        pass
```
Comment (andamian) [inline caom2pipe/transfer_composable.py:104]: Probably a raise NotImplemented could be used here.

**Review by andamian — APPROVED** [review]

File: caom2pipe/execute_composable.py (hunk 1 above)
Comment (SharonGoliath) [reply caom2pipe/execute_composable.py:433]: Given that the execute methods now don't care where the data is stored, knowing it's CADC storage name is pointless, so this commented out code will be removed.

File: caom2pipe/execute_composable.py (hunk 2 above)
Comment (SharonGoliath) [reply caom2pipe/execute_composable.py:1145]: Not needed so removed.

File: caom2pipe/manage_composable.py (hunk 3 above)
Comment (SharonGoliath) [reply caom2pipe/manage_composable.py:437]: Replaced with os.path.join. Tried to catch all the instances in the file.

File: caom2pipe/transfer_composable.py (hunk 4 above)
Comment (SharonGoliath) [reply caom2pipe/transfer_composable.py:104]: This is the default implementation, if the config has use_local_files set to True, so a 'raise' here would unexpectedly stop execution.

**Pull request merged**